src/state.cairo

```diff
@@ -46,22 +50,34 @@ pub struct Header {
     pub nonce: u32,
 }
 
-/// Transaction
-/// https://developer.bitcoin.org/reference/transactions.html#raw-transaction-format
+/// Extended transaction.
+/// https://learnmeabitcoin.com/technical/transaction/
+///
+/// Contains additional "meta" fields required for validation.
 #[derive(Drop, Copy)]
 pub struct Transaction {
     /// The version of the transaction.
     pub version: i32,
+    /// Flag which indicates the presence of witness data.
+    /// Segwit marker and flag do not contribute to TXID (transaction hash),
+    /// but do contribute to wTXID.
+    pub is_segwit: bool,
```

Comment:
your comment talks about `marker` and `flag` (which are 2 different fields in the transaction : 
https://learnmeabitcoin.com/technical/transaction/#structure-marker) , but you use a `is_segwit` field , it might be a bit confusing

but indeed it seems they can be combined (at least for now) in your `is_segwit`field,  maybe adding a comment that in the future, we may need to separate this field in 2 distinct ones, with `flag` incremented for transaction struct updates in the protocol

Reply:
Good point!
I will expand on this in the comment